src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -365,6 +419,169 @@ export default class TableComponent extends ComponentBase implements IComponent,
 		}
 	}
 
+	private registerRowDetailPlugin(settings: azdata.RowDetailsOptions) {
+		const rowDetail = new RowDetailView({
+			cssClass: '_detail_selector',
+			process: (item) => {
+				(<any>rowDetail).onAsyncResponse.notify({
+					'itemDetail': item,
+				}, undefined, this);
+			},
+			useRowClick: true,
+			panelRows: settings.panelRows,
+			postTemplate: (itemDetail) => itemDetail[settings.dataPropertyName],
+			preTemplate: () => '',
+			loadOnce: true,
+			detailsHtml: settings.detailsHtml,
+			headerCssClass: settings.headerCssClass,
+			idGetter: (item) => item.__id__,
+			idSetter: (item, val) => item.__id__ = val
+		});
+		this._rowDetail = rowDetail;
+		this._table.registerPlugin(<any>this._rowDetail);
+	}
+
+
+
+	private registerFilterPlugin() {
+		const filterPlugin = new HeaderFilter<Slick.SlickData>();
+		this._register(attachButtonStyler(filterPlugin, this.themeService));
+		this._filterPlugin = filterPlugin;
+		this._filterPlugin.onFilterApplied.subscribe((e, args) => {
+			let filterValues = (<any>args).column.filterValues;
+			if (filterValues) {
+				this._tableData.refresh();
+				this._table.grid.resetActiveCell();
+			}
+		});
+
+		this._filterPlugin.onCommand.subscribe((e, args: any) => {
+			this.columnSort(args.column.field, args.command === 'sort-asc');
+		});
+
+		filterPlugin['getFilterValues'] = this.getFilterValues;
+		filterPlugin['getAllFilterValues'] = this.getAllFilterValues;
+		filterPlugin['getFilterValuesByInput'] = this.getFilterValuesByInput;
+		this._table.registerPlugin(filterPlugin);
+
+	}
+
+	private columnSort(field: string, isAscending: boolean) {
+		this._tableData.sort((item1, item2) => {
+			return item1[field].toString().localeCompare(item2[field].toString());
+		}, isAscending);
+	}
+
+	private filter(item: any) {
+		let columns = this._table.grid.getColumns();
+		let value = true;
+		for (let i = 0; i < columns.length; i++) {
+			let col: any = columns[i];
+			let filterValues = col.filterValues;
+			if (filterValues && filterValues.length > 0) {
+				if (item._parent) {
+					value = value && find(filterValues, x => x === item._parent[col.field]);
+				} else {
+					let colValue = item[col.field];
+					if (colValue instanceof Array) {
+						value = value && find(filterValues, x => colValue.indexOf(x) >= 0);
+					} else {
+						value = value && find(filterValues, x => x === colValue);
+					}
+				}
+			}
+		}
+		return value;
+	}
+
+	private getFilterValues(dataView: Slick.DataProvider<Slick.SlickData>, column: Slick.Column<any>): Array<any> {
+		const seen: Array<string> = [];
+		for (let i = 0; i < dataView.getLength(); i++) {
+			const value = dataView.getItem(i)[column.field!];
+			if (value instanceof Array) {
```

Comment:
This is for the case when the column contains array as a data object. It works well for just displaying purposes for the grid (it will print comma separated values). But for header filter I need to display distinct values of the array item. 
And I agree that header filter is a better place for that. I will move there.